### ISSUE TITLE
restarting nvda after addon installation is finished.

### ISSUE THREAD

**Issue by miki123211** [nvaccess/nvda#6600]
For now, after an addon is installed, nvda asks the user whether he wants to restart the program now. This should be configurable. I propose to add a combo box in the "general preferences" dialog that would allow the user to choose how NVDA behaves in this situation. The user should be able to choose to always restart, to never restart or to ask after each installation. Possible scenarios where it might help:
When installing NVDA on a new computer, an experienced user often installs a lot of addons at once and he doesn't want nvda to be restarted every time. This feature would speed up the installation process.
When developing an addon, we want nvda to restart after each installation. If this feature was available, it would be much quicker to do so.

**Comment by jcsteh** [comment]
If you use the Install button in the Add-ons Manager, NVDA should only ask you after you close the Add-ons Manager. That is, you can install multiple add-ons without being asked to restart. Only when you close the Add-ons Manager will NVDA ask.

If you install add-ons from Explorer or the like, NVDA has no way of knowing that you're potentially installing more than one add-on, so we can't achieve this.

I don't think having an option to never restart is a good idea, since a user might then be confused as to why the add-on doesn't work yet. They won't necessarily understand that a restart is always required even if they choose never. Similarly, I don't think always restarting is a good idea, since forcing restarts without explicit user action is bound to cause frustration. You can argue that the user configured it and that therefore they want this, but I'm concerned that users won't understand the impact of this choice and it will have impact long after they configured it and perhaps even forgot about configuring it.

We do appreciate the suggestion, but given the above, I'm closing this as something we won't do.
